controllers/openstackbaremetalset_controller.go

```diff
@@ -1429,10 +1434,6 @@ func (r *OpenStackBaremetalSetReconciler) createNewHostnames(
 			return newHostnames, err
 		}
 
-		if instance.Status.BaremetalHosts == nil {
-			instance.Status.BaremetalHosts = map[string]ospdirectorv1beta1.HostStatus{}
-		}
-
```

Comment:
@abays now I understood why this was required :)

Reply:
Ah, was it a necessary to try to avoid the problem you are fixing in this PR?

Reply:
I am not sure I get the question correct, but if we do not fix it you basically can not trust parameters are set correct at places in the reconcile loop and have to validate them somehow. so I think the correct way is to fix it like in this PR